core/src/main/java/org/apache/calcite/sql/parser/SqlParser.java

```diff
@@ -242,64 +245,72 @@ public static ConfigBuilder configBuilder(Config config) {
   /**
    * Interface to define the configuration for a SQL parser.
    */
+  @Value.Immutable(singleton = true)
+  @Value.Style(typeImmutable = "ImmutableSqlParserConfig")
   public interface Config {
     /** Default configuration. */
-    Config DEFAULT = ImmutableBeans.create(Config.class)
-        .withLex(Lex.ORACLE)
-        .withIdentifierMaxLength(DEFAULT_IDENTIFIER_MAX_LENGTH)
-        .withConformance(SqlConformanceEnum.DEFAULT)
-        .withParserFactory(SqlParserImpl.FACTORY);
-
-    @ImmutableBeans.Property()
-    @ImmutableBeans.IntDefault(DEFAULT_IDENTIFIER_MAX_LENGTH)
-    int identifierMaxLength();
+    Config DEFAULT = ImmutableSqlParserConfig.of();
 
+    @Value.Default default int identifierMaxLength() {
+      return DEFAULT_IDENTIFIER_MAX_LENGTH;
+    }
+    //Quoting.DOUBLE_QUOTE, Casing.TO_UPPER, Casing.UNCHANGED, true,
+    //      CharLiteralStyle.STANDARD
```

Comment:
This looks odd

Reply:
Good catch. That was a dev note that should have been deleted. Removed.